Dockerfile

```diff
@@ -1,40 +1,62 @@
 
-WORKDIR /home/ubuntu/eval-dev-quality
-COPY ./ ./
-RUN chown -R ubuntu:ubuntu ./
+# Non-root ollama need a hardcoded directory to store ssh-key.
+RUN mkdir -p /.ollama && chmod 777 /.ollama
+# Same for symflower
+RUN mkdir -p /.config && chmod 777 /.config
+RUN mkdir -p /.eval-dev-quality && chmod 777 /.eval-dev-quality
+RUN mkdir -p /.cache && chmod 777 /.cache
 
+# Switch to the ubuntu user as we want it to run as non-root.
 USER ubuntu
-RUN mkdir -p ~/.eval-dev-quality
+WORKDIR /app
+COPY --chown=ubuntu:ubuntu ./testdata ./testdata
+COPY --chown=ubuntu:ubuntu ./Makefile ./Makefile
+RUN mkdir -p .eval-dev-quality
 
 # Install Maven
 RUN wget https://archive.apache.org/dist/maven/maven-3/3.9.1/binaries/apache-maven-3.9.1-bin.tar.gz && \
-	tar -xf apache-maven-3.9.1-bin.tar.gz -C ~/.eval-dev-quality/ && \
+	tar -xf apache-maven-3.9.1-bin.tar.gz -C /app/.eval-dev-quality/ && \
 	rm apache-maven-3.9.1-bin.tar.gz
-ENV PATH="${PATH}:/home/ubuntu/.eval-dev-quality/apache-maven-3.9.1/bin"
+ENV PATH="${PATH}:/app/.eval-dev-quality/apache-maven-3.9.1/bin"
 
 # Install Gradle
 RUN wget https://services.gradle.org/distributions/gradle-8.0.2-bin.zip && \
-	unzip gradle-8.0.2-bin.zip -d ~/.eval-dev-quality/ && \
+	unzip gradle-8.0.2-bin.zip -d /app/.eval-dev-quality/ && \
 	rm gradle-8.0.2-bin.zip
-ENV PATH="${PATH}:/home/ubuntu/.eval-dev-quality/gradle-8.0.2/bin"
+ENV PATH="${PATH}:/app/.eval-dev-quality/gradle-8.0.2/bin"
 
 # Install Java
 RUN wget https://corretto.aws/downloads/latest/amazon-corretto-11-x64-linux-jdk.tar.gz && \
-	tar -xf amazon-corretto-11-x64-linux-jdk.tar.gz -C ~/.eval-dev-quality/ && \
+	tar -xf amazon-corretto-11-x64-linux-jdk.tar.gz -C /app/.eval-dev-quality/ && \
 	rm amazon-corretto-11-x64-linux-jdk.tar.gz
-ENV JAVA_HOME="/home/ubuntu/.eval-dev-quality/amazon-corretto-11.0.23.9.1-linux-x64"
+ENV JAVA_HOME="/app/.eval-dev-quality/amazon-corretto-11.0.23.9.1-linux-x64"
 ENV PATH="${PATH}:${JAVA_HOME}/bin"
 
 # Install Go
 RUN wget https://go.dev/dl/go1.21.5.linux-amd64.tar.gz && \
-	tar -xf go1.21.5.linux-amd64.tar.gz -C ~/.eval-dev-quality/ && \
+	tar -xf go1.21.5.linux-amd64.tar.gz -C /app/.eval-dev-quality/ && \
 	rm go1.21.5.linux-amd64.tar.gz
-ENV PATH="${PATH}:/home/ubuntu/.eval-dev-quality/go/bin"
-ENV PATH="${PATH}:/home/ubuntu/go/bin"
+ENV PATH="${PATH}:/app/.eval-dev-quality/go/bin"
+RUN go env -w GOPATH=/app/.eval-dev-quality/go
 
-# Setup the evaluation
+# Install the binary
+COPY --from=builder --chown=ubuntu:ubuntu /app/eval-dev-quality /app/.eval-dev-quality/bin/
+ENV PATH="${PATH}:/app/.eval-dev-quality/bin"
+RUN go install -v github.com/vektra/mockery/v2@v2.40.3 && go install -v gotest.tools/gotestsum@v1.11.0
+RUN eval-dev-quality install-tools --install-tools-path /app/.eval-dev-quality/bin
 
-RUN make install-all
-ENV PATH="${PATH}:/home/ubuntu/.eval-dev-quality/bin"
+# CHMOD everything because of non-root user ids
```

Comment:
Bit inconvenient that we install here manually instead of using the makefile. Cause then we have to maintain two things. IS there a way to refactor the makefile to have everything there still?

Reply:
Split it up into two commands where one of it takes an argument, building the container works as well as running `make install-all`